### ISSUE TITLE
binder: encapsulate lower-case conversion for identifiers

### ISSUE THREAD

**Issue by a9QrX3Lu** [risinglightdb/risinglight#285]
According to [this comment](https://github.com/singularity-data/risinglight/pull/283#pullrequestreview-842342347), we should encapsulate the lower-case conversion for better extensibility. However, we can not do this by modifying the Catalog, because we have other states owning the identifiers, e.g., every field in [`BindContext`](https://github.com/singularity-data/risinglight/blob/main/src/binder/mod.rs#L66-L78).

I think the best way to do this is to visit the AST generated by sqlparser, and convert all the names into lower-case.

**Comment by yinfredyue** [comment]
Conceptually, traversing the AST sounds easy and intuitive. But since the AST is implemented as `enum`'s in sqlparser, how do you traverse enums? Manual `match`es?

**Comment by skyzh** [comment]
Yes, that's true 🤣 btw I just realized that Postgres parser will convert everything to lowercase, so we might follow its convention and just convert everything to lowercase.

**Comment by yinfredyue** [comment]
> so we might follow its convention and just convert everything to lowercase.

Do you mean after converting names in the AST?  Converting the query string into lowercase cannot handle the following correctly:

![image](https://user-images.githubusercontent.com/41224888/191164956-51488b61-c9e4-435a-9cc6-2fe4a7932de6.png)

**Comment by skyzh** [comment]
Oh, you're right. I mean all identifiers (like table name, column name) are lowercase. Probably we only need to make all table / column case-insensitive in binder.
